### ISSUE TITLE
Will FLIF be submitted to standardization?

### ISSUE THREAD

**Issue by mzso** [FLIF-hub/FLIF#425]
Hi!

I wonder if FLIF will be pushed to become a standard.? (IETF, ISO, ITU, etc)

I feel like it would have a better chance to gain popularity.

**Comment by SaschaNaz** [comment]
This depends on #162, not sure about the progress...

**Comment by mzso** [comment]
[**@SaschaNaz**](https://github.com/SaschaNaz) commented on [2017. máj. 28. 21:30 CEST](https://github.com/FLIF-hub/FLIF/issues/425#issuecomment-304535066 "2017-05-28T19:30:49Z - Replied by Github Reply Comments"):
> This depends on [#162](https://github.com/FLIF-hub/FLIF/issues/162), not sure about the progress...

I see. There doesn't seem to be any activity though.
It's weird that the owner is not interested since it seems to me one of the key elements of getting the format popularized.  (Right after the fromat actually being good.)

**Comment by jonsneyers** [comment]
The spec at http://flif.info/spec.html is a good start, but it's not finished yet. Mostly the transforms still need to be written out.

I am interested in making FLIF a standard, but I'm not at all familiar with the process. Maybe someone who has experience in standardization could help us out?

**Comment by maddin200** [comment]
Hi,
every new standard starts as a proposal in the right ISO group - here probably 
ISO/IEC JTC 1/SC 29
Coding of audio, picture, multimedia and hypermedia information
The standard has a different status from Working Draft to Final Comittee Draft. This process can take a while.
Bye Martin

**Comment by est31** [comment]
I have no experience with standardisation, but generally ISO puts its published standards behind a (big) paywall, while other standardisation bodies like the IETF make their standards available free of charge, which greatly helps if you just want to look up something about the standard.

**Comment by mzso** [comment]
[**@jonsneyers**](https://github.com/jonsneyers) commented on [2017. jún. 7. 16:12 CEST](https://github.com/FLIF-hub/FLIF/issues/425#issuecomment-306807427 "2017-06-07T14:12:43Z - Replied by Github Reply Comments"):
> If we try to formally standardize FLIF, I think it would make a lot of sense to ensure that the standard does not end up behind a paywall – FLIF is not Free as in beer ;) but I see no reason for its standard to be non-free as in beer :)
> 
> Would it make sense to (try to) make it a W3C standard? Or maybe try to get it recognized somehow by the Alliance for Open Media ([http://aomedia.org/](http://aomedia.org/)) or Xiph.org ([https://www.xiph.org/](https://www.xiph.org/)) ?

Any barrier to have it submitted to multiple standardization authorities?
I see most formats are standardized by more than one authority. (Jpeg: ISO, IEC, ITU, PNG: ISO, IEC IETF)
I wonder how they did it.
(For MPEG-4 AVC wikipedia claims that the ITU/ISO standard is jointly maintained.)

**Comment by est31** [comment]
I'm not sure about W3C's scope, maybe FLIF falls inside it, no idea.

About the Alliance for Open Media and Xiph, contacting them may never harm, but I guess that both are more actual places which create standards themselves, not who publish them in a standardized (:p) way. 

Xiph themselves have submitted opus to the IETF for standardization, where the official place is now. The standards they did themselves were very inconsistently published: The Theora spec is a PDF, the ogg and vorbis specs are html, and the OGG skeleton spec is even just a wiki page. Some of it has been "standardized" by the IETF since, but only as informal document, not actual standard.

**Comment by mzso** [comment]
[**@est31**](https://github.com/est31) commented on [2017. jún. 7. 16:46 CEST](https://github.com/FLIF-hub/FLIF/issues/425#issuecomment-306817926 "2017-06-07T14:46:45Z - Replied by Github Reply Comments"):
> Xiph themselves have submitted opus to the IETF for standardization, where the official place is now. The standards they did themselves were very inconsistently published: The Theora spec is a PDF, the ogg and vorbis specs are html, and the OGG skeleton spec is even just a wiki page. Some of it has been "standardized" by the IETF since, but only as informal document, not actual standard.

This probably didn't help them achieve popularity, which they didn't...


> About the Alliance for Open Media and Xiph, contacting them may never harm, but I guess that both are more actual places which create standards themselves, not who publish them in a standardized (:p) way.
 
Well, it's not like AV1 was custom designed by AOM. It manly consists of different parts of codecs that some of the members were developing themselves before the alliance. They don't seem to oppose building on pre-existing stuff. Which they aim for IETF's NETVC.

IETF seems to be the most relevent since as an image format probably browsers are the largest market. (Not sure if FLIF is suitable for hardware encoding and digital cameras or not.)

**Comment by maddin200** [comment]
Hi,
ISO is not automatically a paywall:
http://standards.iso.org/ittf/PubliclyAvailableStandards/index.html
I must ask a colleague what the conditions are.
Bye Martin

**Comment by thebombzen** [comment]
The ISO/IEC standards are often paywalled, but amusingly, you can find them elsewhere. In particular, the spec for MPEG-4 AVC is behind a paywall, but the H.264 can be freely downloaded from ITU's website.

**Comment by maddin200** [comment]
As far as I know everybody can make a proposal for an ISO standard with a application at the local standard organization (in Germany DIN and so on).

**Comment by SaschaNaz** [comment]
>Would it make sense to (try to) make it a W3C standard?

PNG already is defined as W3C standard, why not FLIF? :smile:

It seems some early-day standards are first grown on [WICG](https://github.com/wicg/) in these days [before being a complete standard](https://github.com/w3c/html/issues/426).

**Comment by jessetrana** [comment]
Any further movement on this? I've been watching this project with great interest for quite some time and I would love to see it turn into a standard - even if that would be a few years out.

**Comment by saschanaz** [comment]
AFAIK no progress. I opened [a post](https://discourse.wicg.io/t/does-wicg-also-receive-image-format-specification/2541/1) on WICG, I hope someone be interested...

**Comment by waldyrious** [comment]
Doesn't this still depend on #162 ("Write out the detailed FLIF stream specification")?

Also, wouldn't the ["affects bitstream"](https://github.com/FLIF-hub/FLIF/issues?q=is%3Aopen+label%3A"affects+bitstream") issues need to be resolved first as well? Most of them are under the ["M3 - Future extensions"](https://github.com/FLIF-hub/FLIF/milestone/3) milestone, but #362 isn't.

**Comment by jessetrana** [comment]
Looks like WICG have started to provide some good feedback about how/where to start, even if it is an uphill battle. I guess a question I have as a newcomer to the community is "Where is FLIF in 5 years ideally?" If standardization is a major desire, that's probably going to be a big, multi-phase push - which may or may not be anyone's passion. But if it's not, it will probably just take time/energy away from the other good stuff happening here.

**Comment by mzso** [comment]
[**@jonsneyers**](https://github.com/jonsneyers) commented on [2017. jún. 7. 16:12 CEST](https://github.com/FLIF-hub/FLIF/issues/425#issuecomment-306807427 "2017-06-07T14:12:43Z - Replied by Github Reply Comments"):
> Would it make sense to (try to) make it a W3C standard? Or maybe try to get it recognized somehow by the Alliance for Open Media ([http://aomedia.org/](http://aomedia.org/)) or Xiph.org ([https://www.xiph.org/](https://www.xiph.org/)) ?

How about joining the alliance? Is there any barrier to that?

They not too long ago expressed interest in creating a still image format anyway.

**Comment by saschanaz** [comment]
AOM is already developing AVIF based on AV1 anyway. https://aomediacodec.github.io/av1-avif/

**Comment by mzso** [comment]
[**@saschanaz**](https://github.com/saschanaz) commented on [2018. jún. 1. 11:52 CEST](https://github.com/FLIF-hub/FLIF/issues/425#issuecomment-393832445 "2018-06-01T09:52:28Z - Replied by Github Reply Comments"):
> AOM is already developing AVIF based on AV1 anyway. [https://aomediacodec.github.io/av1-avif/](https://aomediacodec.github.io/av1-avif/)

That's a lossy format though. If it can do lossless I doubt that it's effective. There's room for a lossless format the way I see it.

**Comment by wingman-jr-addon** [comment]
@jonsneyers I was excited to see your announcement about JPEG XL, which sounds rather like the standards-based successor to FLIF, in a roundabout way. I'm guessing that means this can be closed?

**Comment by ziemek99** [comment]
> I'm guessing that means this can be closed?

FLIF has been superseded by JPEG XL, and yes, that means this issue can be closed.

You can read more about JPEG XL on an [official website](https://jpeg.org/jpegxl/), [community website](https://jpegxl.info), dig into [GitHub repository](https://github.com/libjxl/libjxl) or join Jon, other JPEG XL devs and the community on a [Discord server](https://discord.gg/DqkQgDRTFu).

**Comment by saschanaz** [comment]
Should this repo be archived?

**Comment by ziemek99** [comment]
No, there are still bugs to be fixed and issues to be answered. It's just not that urgent. Fortunately it won't consume Jon's precious time (which is already reserved for JPEG XL, his private life and other things) because I can now manage this repo's issues, PRs and releases. Personally I prefer to keep FLIF *retired* instead of *archived*, I don't like ending unfinished things. Maybe I'll finish exactly on a 1000th commit (or some other nice round number)? 😜

FUIF repo, on the other hand, should be archived.
